pyproject.toml

```diff
@@ -30,6 +30,7 @@ dependencies = [
     "sentence-transformers>=3.0.1,<4.0.0",
     "tiktoken>=0.7.0,<1.0.0",
     "tqdm>=4.66.4,<5.0.0",
+    "lxml>=5.3.0"
```

Comment:
I think its better to not require `lxml` since its only needed if a user wants to use `SerializationFormat::XML`. What if, during the serialization code, we do something like

```
    elif lotus.settings.serialization_format == SerializationFormat.XML:
        try:
            from lxml import etree
        except ImportError:
            raise ImportError(
                "The 'lxml' library is required for XML serialization. "
                "You can install it with the following command:\n\n"
                "    pip install 'lotus-ai[xml]'"
            )
```

And then in the `pyproject.toml` we have it as an optional dependency
```
[project.optional-dependencies]
docs = [
    "sphinx",
    "sphinx_rtd_theme",
]
xml = [
    "lxml",
]
```

Reply:
Done